controllers/terminal_controller.go

```diff
@@ -1329,6 +1360,54 @@ func NewClientSetFromSecret(config *rest.Config, secret *corev1.Secret, opts cli
 	return nil, errors.New("no valid kubeconfig found")
 }
 
+func NewClientSetFromGoogleSAKey(ctx context.Context, kubeconfig []byte, gsaKey []byte, opts client.Options) (*ClientSet, error) {
+	clientConfig, err := clientcmd.NewClientConfigFromBytes(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg, err := clientConfig.RawConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	context := cfg.Contexts[cfg.CurrentContext]
+	if context == nil {
+		return nil, fmt.Errorf("no context found for current context %s", cfg.CurrentContext)
+	}
+
```

Comment:
Couldn't the context from L1325 be passed to this method to save the steps to retrieve it again?

Reply:
sure, will change it